### PR TITLE
Amanda axios setup

### PR DESCRIPTION
**Pull request by phillipsdev**
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change
Server setup for user registration and login

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested on Postman

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,14 +1,25 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { Component } from "react";
+import logo from "./logo.svg";
+import "./App.css";
+import axios from "axios";
 
 class App extends Component {
+  state = {
```
Comment (rusticlion) [inline client/src/App.js:7]: This is purely to demonstrate end-to-end functionality. The next revision will encapsulate all HTTP requests within Redux action creators.

File: client/src/index.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
-import ReactDOM from 'react-dom';
-import './index.css';
-import App from './App';
-import registerServiceWorker from './registerServiceWorker';
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+import App from "./App";
+import registerServiceWorker from "./registerServiceWorker";
+import { createStore } from "redux";
```
Comment (rusticlion) [inline client/src/index.js:6]: Redux imports are in place to configure the store and provider in the next revision.

File: server/api/controllers/UserControllers.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+const User = require("../models/User");
+const jwt = require("jsonwebtoken");
+require("dotenv").config();
+
+const secret = process.env.SECRET;
+
+function generateToken(user) {
```
Comment (rusticlion) [inline server/api/controllers/UserControllers.js:7]: JWTs are generated on successful register or login and returned to the user in the HTTP response.

File: server/api/controllers/UserControllers.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  User.findOne({ username: username.toLowerCase() })
+    .then(user => {
+      user
+        .checkPassword(password)
+        .then(success => {
+          res.status(200);
+          const token = generateToken(user);
+          res.json({ username, token });
+        })
+        .catch(err => {
```
Comment (rusticlion) [inline server/api/controllers/UserControllers.js:44]: Note the use of separate catch blocks, statuses, and messages for different errors.

File: server/api/routes.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const { register, login } = require("./controllers/UserControllers");
```
Comment (rusticlion) [inline server/api/routes.js:1]: This routing module is invoked on the Express app in `server.js` to wire request routes to the appropriate controllers.

File: server/server.js [hunk 6]
```diff
@@ -12,6 +23,8 @@ mongoose.connect(
     console.log("connected to mongo");
   }
 );
+mongoose.set("useCreateIndex", true);
```
Comment (rusticlion) [inline server/server.js:26]: Future-proofed Mongoose configuration - the default is soon-to-be deprecated.

**Review by rusticlion — APPROVED** [review]
Registration and login are functioning on the API, talking to the live DB and providing JWTs. At this point, testing has used Postman - next on deck is the front-end buildout. This code was mob programmed by the full team, led by Amanda.

**Review by emukupa — APPROVED** [review]
Excellent Job guys! Keep it up:)

**Pull request merged**